core/commands/keystore.go

```diff
@@ -143,6 +151,159 @@ func formatID(id peer.ID, formatLabel string) string {
 	}
 }
 
+var keyExportCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Export a keypair",
+		ShortDescription: `
+Exports a named libp2p key to disk.
+
+By default, the output will be stored at './<key-name>', but an alternate
+path can be specified with '--output=<path>' or '-o=<path>'.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("name", true, false, "name of key to export").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption(outputOptionName, "o", "The path where the output should be stored."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		name := req.Arguments[0]
+
+		if name == "self" {
+			return fmt.Errorf("cannot export key with name 'self'")
```

Comment:
Hm. Is there any reason not to allow this?

Reply:
I guess not, but we have to the keys live in different places and it's a little messier. Let's add it in a followup PR